Rubberduck.Parsing/Symbols/IdentifierReference.cs

```diff
@@ -81,28 +81,31 @@ public bool HasExplicitCallStatement()
             return Context.Parent is VBAParser.CallStmtContext && ((VBAParser.CallStmtContext)Context).CALL() != null;
         }
 
-        private Lazy<bool> _hasTypeHint = new Lazy<bool>(() => ComputeHasTypeHint());
```

Comment:
I'm pretty sure you don't need the line 84-91. You only need one line here -- `private bool HasTypeHint() => new Lazy<bool>(() => ComputeTypeHint()).Value;` (untested aircode!)

Reply:
not correct. that's not caching the value again. What's necessary is initializing `_hasTypeHint` in the ctor. @bclothier your code instantiates a new Lazy every invocation of `HasTypeHint()`, which is exactly what we wanted to avoid.

HasTypeHint should become: `public bool HasTypeHint() => _hasTypeHint.Value;`